File: citame_back/src/models/business.model.js
const { Schema, model } = require('mongoose')
const mongoose = require('mongoose')

//Datos que se guardan en la BD
const businessSchema = new Schema({
  businessName: { type: String, required: true },
  category: { type: String, required: true },
  email: { type: String, required: true },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'usuario',
    required: true,
  },
  workers: [String],
  contactNumber: { type: String, required: true },
  direction: { type: String, required: true },
  latitude: { type: String, required: true },
  longitude: { type: String, required: true },
  description: { type: String, required: true },
  imgPath: [{ type: Schema.Types.ObjectId, ref: 'Imagen' }],
  horario: { type: String },
  servicios: [{ type: mongoose.Schema.Types.ObjectId, ref: 'services' }],
  citas: [{ type: Schema.Types.ObjectId, ref: 'cita' }],
})

module.exports = model('business', businessSchema)

File: citame/lib/models/business_model.dart
import 'dart:convert';

class Business {
  final String businessName;
  final String idMongo;
  final String category;
  final String email;
  //final String createdBy;
  final List<dynamic> workers;
  final String contactNumber;
  final String direction;
  final String latitude;
  final String longitude;
  final String description;
  final List imgPath;
  final Map horario;
  Map toJson() => {
        'businessName': businessName,
        'idMongo': idMongo,
        'category': category,
        'email': email,
        'workers': workers,
        'contactNumber': contactNumber,
        'direction': direction,
        'latitude': latitude,
        'longitude': longitude,
        'description': description,
        'imgPath': imgPath,
        'horario': horario,
      };
  Business({
    required this.businessName,
    required this.idMongo,
    required this.category,
    required this.email,
    //required this.createdBy,
    required this.workers,
    required this.contactNumber,
    required this.direction,
    required this.latitude,
    required this.longitude,
    required this.description,
    required this.imgPath,
    required this.horario,
  });

  factory Business.fromJson(Map<String, dynamic> json) {
    return Business(
      businessName: json['businessName'],
      idMongo: json['_id'],
      category: json['category'],
      email: json['email'],
      //createdBy: json['createdBy'],
      workers: json['workers'],
      contactNumber: json['contactNumber'],
      direction: json['direction'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      description: json['description'],
      imgPath: json['imgPath'],
      horario: jsonDecode(json['horario']),
    );
  }
  factory Business.fromJson2(Map<String, dynamic> json) {
    return Business(
      businessName: json['businessName'],
      idMongo: json['idMongo'],
      category: json['category'],
      email: json['email'],
      //createdBy: json['createdBy'],
      workers: json['workers'],
      contactNumber: json['contactNumber'],
      direction: json['direction'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      description: json['description'],
      imgPath: json['imgPath'],
      horario: jsonDecode(json['horario']),
    );
  }
}
